lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -242,6 +243,9 @@ def __get_k8s_job_spec_template(self, ajs):
         # TODO include other relevant elements that people might want to use from
         # TODO http://kubernetes.io/docs/api-reference/v1/definitions/#_v1_podspec
         k8s_spec_template["spec"]["securityContext"] = self.__get_k8s_security_context()
+        extra_metadata = yaml.safe_load(self.runner_params['k8s_job_metadata'] or "{}")
```

Comment:
This has to be an anti-pattern at this point - we've had YAML job confs for a while and that I implemented explicitly so we aren't loading JSON from XML fragments. Any chance we can at least only do the safe load if it is a string and not a dictionary?

Reply:
Maybe make map above an identity of some sort?